index.d.ts

```diff
@@ -0,0 +1,1353 @@
+     * UUID path. Latest value of characteristic may not be stored inside returned object.
+     */
+    writeCharacteristicWithResponse(
+      characteristicUUID: UUID,
+      valueBase64: Base64,
+      transactionId?: string
+    ): Promise<Characteristic>;
+
+    /**
+     * {@link #blemanagerwritecharacteristicwithoutresponsefordevice|bleManager.writeCharacteristicWithoutResponseForDevice()} with partially filled arguments.
+     *
+     * @param {UUID} characteristicUUID {@link Characteristic} UUID.
+     * @param {Base64} valueBase64 Value in Base64 format.
+     * @param {?TransactionId} transactionId optional `transactionId` which can be used in
+     * {@link #blemanagercanceltransaction|bleManager.cancelTransaction()} function.
+     * @returns {Promise<Characteristic>} Promise which emits first {@link Characteristic} object matching specified
+     * UUID path. Latest value of characteristic may not be stored inside returned object.
+     */
+    writeCharacteristicWithoutResponse(
+      characteristicUUID: UUID,
+      valueBase64: Base64,
+      transactionId?: string
+    ): Promise<Characteristic>;
+
+    /**
+     * {@link #blemanagermonitorcharacteristicfordevice|bleManager.monitorCharacteristicForDevice()} with partially filled arguments.
+     *
+     * @param {UUID} characteristicUUID - {@link Characteristic} UUID.
+     * @param {function(error?: BleError, characteristic?: Characteristic)} listener callback which emits
+     * {@link Characteristic} objects with modified value for each notification.
+     * @param {?TransactionId} transactionId optional `transactionId` which can be used in
+     * {@link #blemanagercanceltransaction|bleManager.cancelTransaction()} function.
+     * @returns {Subscription} Subscription on which `remove()` function can be called to unsubscribe.
+     */
+    monitorCharacteristic(
+      characteristicUUID: UUID,
+      listener: (error?: BleError, characteristic?: Characteristic) => void,
+      transactionId?: string
+    ): Subscription;
+  }
+
+  /**
+   * Native device object passed from BleModule.
+   * @private
+   */
+  export interface NativeDevice {
+    /**
+     * Device identifier: MAC address on Android and UUID on iOS.
+     * @private
+     */
+    id: Device['id'];
+    /**
+     * Device name if present
+     * @private
+     */
+    name?: string;
+    /**
+     * Current Received Signal Strength Indication of device
+     * @private
+     */
+    rssi?: number;
+    /**
+     * Current Maximum Transmission Unit for this device. When device is not connected
+     * default value of 23 is used.
+     * @private
+     */
+    mtu: number;
+
+    // Advertisement
+
+    /**
+     * Device's custom manufacturer data. Its format is defined by manufacturer.
+     * @private
+     */
+    manufacturerData?: Base64;
+
+    /**
+     * Map od service UUIDs with associated data.
+     * @private
+     */
+    serviceData?: { [uuid: UUID]: Base64 };
```

Comment:
`serviceData: { [uuid: string]: Base64 } | null;`

Reply:
We need to use `string` here directly because type aliases are not accepted as indices.